### PR TITLE
Daniel/anonymousstate

### PR DESCRIPTION
**Pull request by illicitonion**
Depends on https://github.com/matrix-org/synapse/pull/345

Tests in https://github.com/matrix-org/sytest/pull/43

### CODE REVIEW

File: synapse/handlers/sync.py [hunk 1]
```diff
@@ -360,11 +366,16 @@ def incremental_sync_with_gap(self, sync_config, since_token):
...
+        room_ids = [room.room_id for room in rooms]
+
         presence_source = self.event_sources.sources["presence"]
-        presence, presence_key = yield presence_source.get_new_events_for_user(
+        presence, presence_key = yield presence_source.get_new_events(
             user=sync_config.user,
             from_key=since_token.presence_key,
             limit=sync_config.filter.presence_limit(),
+            room_ids=room_ids,
+            is_guest=False,
```
Comment (NegativeMjark) [inline synapse/handlers/sync.py]: Why is "is_guest" set to false here?

File: synapse/handlers/sync.py [hunk 2]
```diff
@@ -312,10 +317,11 @@ def ephemeral_by_room(self, sync_config, now_token, since_token=None):
         receipt_key = since_token.receipt_key if since_token else "0"
 
         receipt_source = self.event_sources.sources["receipt"]
-        receipts, receipt_key = yield receipt_source.get_new_events_for_user(
+        receipts, receipt_key = yield receipt_source.get_new_events(
             user=sync_config.user,
             from_key=receipt_key,
             limit=sync_config.filter.ephemeral_limit(),
+            room_ids=room_ids,
```
Comment (NegativeMjark) [inline synapse/handlers/sync.py]: Why doesn't this have a "is_guest=False"?

File: synapse/handlers/sync.py (hunk 1 above)
Comment (illicitonion) [inline synapse/handlers/sync.py]: Added comment.

File: synapse/handlers/sync.py (hunk 2 above)
Comment (illicitonion) [inline synapse/handlers/sync.py]: It should. It does now. This implies that there is zero test coverage for this method.

Done

**Comment by illicitonion** [comment]
Accidentally pushed as ca2f907 in a squash-merge of another PR... Oops...

**Pull request closed without merging**